walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,124 @@
+
+        // Array of queries represenitng times to meet.
+        List<TimeRange> query = new ArrayList<>();
+        
+        int numberOfEvents = events.size();
+        
+        int timeOfStart = START_OF_DAY;
+        
+        int desiredDuration = (int) request.getDuration();
+        
+        //  Event counter used whilst iterating.
+        int eventCounter = 0;
+        
+        int eventStart = 0;
+        int eventEnd = 0;
+
+        // Set of stirngs representing unique employees set to attend this meeting.
+        Set<String> attendees = new HashSet();
+
+        // Collection of string reprenting who is to attend the requedted meeting
+        Collection<String> requestAttendees = request.getAttendees();
+        
+        // Boolean used to indictae whter or not the attendee should be ignored.
+        boolean shouldIgnore = true;
+
+        // Check if duration of event is longer than a whole day and if so return empty.
+        if ((int)request.getDuration() > WHOLE_DAY.duration()) {
+            return query;
+        }
+
+        // Check if there are any events happening at all if so return whole day query.
+        if (numberOfEvents == 0) {
+            // Adds a 24 hour time range to query.
+            query.add(TimeRange.WHOLE_DAY);
+            return query;
+        }
+
+        // Iterating throguh input event collection. 
+        for (Event event : events) {
+
+            // Collection of the pople who are to attend meeting.
+            attendees = event.getAttendees();
+
+            // Ignore event only if attendees of this event are not needed in requested meeting
+            for (String attendee : attendees) {
+                // If this attendee is required
+                if (requestAttendees.contains(attendee)) {
+                    // You shouldn't ignore this attendee. 
+                    shouldIgnore = false;
+                    break;
+                } 
+                // If at the last event.
+                if (eventCounter == numberOfEvents-1) {
+                    // Add time range of current start time to end of day.
+                    query.add(TimeRange.fromStartEnd(timeOfStart, END_OF_DAY, /* inclusive= */ true));
+                }
+            }
+            // if the attendee is mandatory.
+            if (!shouldIgnore) {
+                
+                // Fetch event start time.
+                eventStart = event.getWhen().start();
+                // Fetch event end time.
+                eventEnd = event.getWhen().end();
+
+
+                //  If the start of the evnt is directly after another or at the start of the day.
+                //  set start of the day to the end of the meeting.
+                if (eventStart == timeOfStart) {
+                    timeOfStart = eventEnd;
+                }
+
+                // Check for overlapping/nested events
+                if (eventStart < timeOfStart) {
```

Comment:
(Not sure why the conversation doesn't apply here after the changes, just in case you missed it)
I think there's a potential problem in this logic here: say eventStart == timeOfStart, so timeOfStart will = eventEnd. Since eventStart < eventEnd, eventStart < timeOfStart is guaranteed but eventEnd > timeOfStart is guaranteed not to happen, so this check seemed to be fine. But I think we should still fix it. The fix should be using another variable to hold the new timeOfStart value, for the next iteration.

Reply:
I'm having trouble understanding